### PR TITLE
Fix NPE in BallerinaSymbol Occurring in Invocation Args Related Type Cast Code Actions

### LINKED ISSUE #32611: NPE in BallerinaSymbol
**Description:**
For the  `incompatible types: expected '([string]|record {| string arg2; |})', found '()'` diagnostic  LS provide the TypeCastCodeAction. Get the following st of NPE when hovered on the squiggle shown for the diagnostic. 

st

Apparently the  NPE is thrown in  the `init` method in  `BallerinaSymbol`. 

**Steps to reproduce:**
```ballerina
function myFunc(){
    MyClass c1 = new(10,<cursor>))

}

class MyClass {
    function init(int arg1, string arg2) {
        
    }
}

```

**Affected Versions:**
slbeta4-snapshot

### PR DESCRIPTION
**Pull request by sanjana**
Fixes #32611

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/DiagnosticsTest.java [hunk 1]
```diff
@@ -79,4 +97,37 @@ private void assertDiagnostic(Diagnostic diagnostic, Object[] expected) {
...
+    }
+
+    private void assertDiagnosticProperties(List<Diagnostic> diagnostics) {
+        int index = 0;
+
+        for (Diagnostic diagnostic : diagnostics) {
+
+            List<DiagnosticProperty<?>> diagnosticProperties = diagnostic.properties();
+
+            if (diagnosticProperties != null) {
```
Comment (nadeeshaan) [inline tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/DiagnosticsTest.java:119]: Let's use early exits. So we can minimize the nested indentations.

**Review by nadeeshaan — CHANGES_REQUESTED** [review]

File: tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/DiagnosticsTest.java (hunk 1 above)
Comment (sanjana) [reply tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/DiagnosticsTest.java]: Updated.

**Review by nadeeshaan — APPROVED** [review]
LGTM

File: tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/DiagnosticsTest.java [hunk 2]
```diff
@@ -79,4 +101,37 @@ private void assertDiagnostic(Diagnostic diagnostic, Object[] expected) {
...
+            }
+
+            for (DiagnosticProperty<?> property : diagnosticProperties) {
+
+                // The diagnostic property value is an instance of TypeSymbol only if the property kind is SYMBOLIC.
+                if (property.kind() == DiagnosticPropertyKind.SYMBOLIC) {
+                    TypeSymbol typeSymbol = (TypeSymbol) property.value();
+                    assertEquals(typeSymbol.signature(), getExpectedDiagnosticProperties()[index][0]);
+                    assertEquals(typeSymbol.typeKind(), getExpectedDiagnosticProperties()[index++][1]);
+                }
```
Comment (anupama-pathirage) [inline tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/DiagnosticsTest.java:130]: Shall we remove these unnecessary new lines?

File: tests/ballerina-compiler-api-test/src/test/resources/test-src/testerrorproject/diagnostic_properties_check.bal [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+function myFunc(){
+    MyClass c1 = new(10,))
+
```
Comment (anupama-pathirage) [inline tests/ballerina-compiler-api-test/src/test/resources/test-src/testerrorproject/diagnostic_properties_check.bal:19]: extra new lines?

File: tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/DiagnosticsTest.java (hunk 2 above)
Comment (sanjana) [reply tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/DiagnosticsTest.java:127]: Removed the superfluous newlines, except for the ones that are surrounding the if block.

File: tests/ballerina-compiler-api-test/src/test/resources/test-src/testerrorproject/diagnostic_properties_check.bal (hunk 3 above)
Comment (sanjana) [reply tests/ballerina-compiler-api-test/src/test/resources/test-src/testerrorproject/diagnostic_properties_check.bal]: Removed the superfluous newlines.

**Review by mohanvive — APPROVED** [review]

**Pull request merged**